src/axom/core/utilities/Annotations.cpp

```diff
@@ -0,0 +1,435 @@
+    return axom::fmt::format("{:%S} seconds:timeval",
+                             std::chrono::seconds {tv->tv_sec} +
+                               std::chrono::microseconds {tv->tv_usec});
+  }
+  case adiak_version:
+    return axom::fmt::format("{}:version", static_cast<char *>(val->v_ptr));
+  case adiak_string:
+    return axom::fmt::format("{}", static_cast<char *>(val->v_ptr));
+  case adiak_catstring:
+    return axom::fmt::format("{}:catstring", static_cast<char *>(val->v_ptr));
+  case adiak_path:
+    return axom::fmt::format("{}:path", static_cast<char *>(val->v_ptr));
+  case adiak_range:
+    return axom::fmt::format("{}",
+                             axom::fmt::join(get_vals_array(t, val, 2), " - "));
+  case adiak_set:
+    return axom::fmt::format(
+      "[{}]",
+      axom::fmt::join(get_vals_array(t, val, adiak_num_subvals(t)), ", "));
+  case adiak_list:
+    return axom::fmt::format(
+      "{{{}}}",
+      axom::fmt::join(get_vals_array(t, val, adiak_num_subvals(t)), ", "));
+  case adiak_tuple:
+    return axom::fmt::format(
+      "({})",
+      axom::fmt::join(get_vals_array(t, val, adiak_num_subvals(t)), ", "));
+  }
+}
+
+static void get_namevals_as_map(const char *name,
+                                int AXOM_UNUSED_PARAM(category),
+                                const char *AXOM_UNUSED_PARAM(subcategory),
+                                adiak_value_t *value,
+                                adiak_datatype_t *t,
+                                void *opaque_value)
+{
+  // add each name/value to adiak metadata map
+  using kv_map = std::map<std::string, std::string>;
+  auto &metadata = *static_cast<kv_map *>(opaque_value);
+  metadata[name] = adiak_value_as_string(value, t);
+}
+
+#endif  // AXOM_USE_ADIAK
+
+}  // namespace detail
+
+#ifdef AXOM_USE_MPI
+void initialize(MPI_Comm comm, const std::string &mode)
+{
+  int num_ranks {1};
+  MPI_Comm_size(MPI_COMM_WORLD, &num_ranks);
+
+  detail::initialize_adiak(comm);
+  detail::initialize_caliper(mode, num_ranks);
+
+  declare_metadata("axom_version", axom::getVersion());
+}
+#endif
+
+void initialize(const std::string &mode)
+{
+  detail::initialize_adiak();
+  detail::initialize_caliper(mode, 1);
+
+  declare_metadata("axom_version", axom::getVersion());
+}
+
+void finalize()
+{
+#ifdef AXOM_USE_ADIAK
+  if(adiak_initialized)
+  {
+    adiak::fini();
+  }
+  adiak_initialized = false;
+#endif
+#ifdef AXOM_USE_CALIPER
+  if(cali_mgr)
+  {
+    for(auto &channel : cali_mgr->get_all_channels())
+    {
+      channel->flush();
+    }
+
+    delete cali_mgr;
+    cali_mgr = nullptr;
+  }
+#endif
+}
+
+void begin(const std::string &name)
+{
+#ifdef AXOM_USE_CALIPER
+  cali::Caliper().begin(
+    cali::region_attr,
+    cali::Variant(CALI_TYPE_STRING, name.c_str(), name.length()));
+#else
+  AXOM_UNUSED_VAR(name);
+#endif
+}
+
+void end(const std::string & /*name*/)
+{
+#ifdef AXOM_USE_CALIPER
+  cali::Caliper().end(cali::region_attr);
+#endif
+}
```

Comment:
The unused argument makes me increasingly uneasy.  Unless there's a strong reason to keep `/*name*/`, please remove it.  The usage pattern is

```
AXOM_ANNOTATE_BEGIN(name);
// many lines of code
AXOM_ANNOTATE_END(name);
```

The danger I see here comes from copying and pasting.  If the user changes the annotation for START but forgets to change the END name to match, any benefit from documenting which block is being ENDed gets replaced with an active harm that is all the more insidious because the error has no immediate effect.  And if at some time down the road we do start to use `/*name*/`, the submarine code errors will surface.

Aside from the stale-comment danger, it's an annoyance to type in redundant information for a required but unused `/*name*/`.

Reply:
And further, using a pattern like

```
AXOM_ANNOTATE_BEGIN(name);
// many lines of code
AXOM_ANNOTATE_END();
```

is much more like the pattern you just introduced, which I think is much nicer:

```
{
AXOM_ANNOTATE_SCOPE(name);
// many lines of code
}
```

Reply:
Some notes:
* I agree about the uneasiness
* Caliper's end doesn't require the name parameter, so it's mostly there as documentation. 
  I think:
```cpp
AXOM_ANNOTATE_END(name);
```
is better than 
```cpp
AXOM_ANNOTATE_END() // end of `name` region
```
which is what we'd likely end up with if we removed the parameter since it makes it easier to pattern match and/or pattern match the code visually and/or w/ `grep`
   * I'd prefer it if caliper was able to take that parameter.
* Caliper has a `validator` service that can be used to check that all the begins have a matching end. I will document this when I update the user docs.

Perhaps this warrants further discussion?

Reply:
Generally, I also prefer a pattern of the form:
```cpp
{
  AXOM_ANNOTATE_SCOPE(name);
  // many lines of code
}
```

However it has a few limitation of varying degrees of complications:
* We cannot use this pattern in cases where variables are introduced within the scope that need to be used outside of it
e.g.
```cpp
  {
    AXOM_ANNOTATE_SCOPE(name);
    int x = <some_value
  }

  // outside scope
  print(x); // ERROR -- x no longer in scope
```

* We can only define one `AXOM_ANNOTATE_SCOPE` within a scope
* It adds an indentation level to the code within the introduced scope

Reply:
I agree that there's a need both for `AXOM_ANNOTATE_SCOPE` and `AXOM_ANNOTATE_BEGIN` / `END`, and that we both prefer `SCOPE`.

In the end, it's your PR so it's your style call.  I think that `AXOM_ANNOTATE_END(); // end of "name" region` is more honest than passing `name` as an argument, because as I already said, here, "name" has exactly the same effect as a comment, but principle of least surprise would say that an argument should do something or be removed.  At least a comment *is* syntactically a comment, and is likely to be highlighted as such.  If you really want a "name" argument to `AXOM_ANNOTATE_END()`, and if Caliper supports it, pull the current block name out of Caliper and compare it to `name` to make sure you're closing the block you think you're closing.  Or even, do that in debug builds and skip the check in release builds.

But in the end, it's your PR so it's your style call.  This particular hill is much smaller than enforcement of 80-character lines or abolition of one-true-brace style.  And I meant it when I approved it.

Reply:
Thanks for the comments @agcapps 

I decided to leave it as is for this PR, but created https://github.com/LLNL/axom/issues/1315 to keep the question open for the future